assets/blocks/featured-video/index.js

```diff
@@ -45,6 +44,36 @@ export default {
 			}
 			previousBlockCount.current = innerBlockCount;
 		}, [ innerBlockCount, clientId, replaceInnerBlocks ] );
+
+		const { parentBlocks, rootClientId, blockIndex } = useSelect(
+			( select ) => {
+				const {
+					getBlockParents,
+					getBlockRootClientId,
+					getBlockIndex,
+				} = select( 'core/block-editor' );
+				return {
+					parentBlocks: getBlockParents( clientId ),
+					rootClientId: getBlockRootClientId( clientId ),
+					blockIndex: getBlockIndex( clientId ),
+				};
+			},
+			[ clientId ]
+		);
+
+		// Move Featured Video block to top at top level.
+		useEffect( () => {
+			if ( parentBlocks?.length || blockIndex ) {
+				moveBlockToPosition( clientId, rootClientId, '', 0 );
+			}
+		}, [
+			parentBlocks,
+			rootClientId,
+			blockIndex,
+			moveBlockToPosition,
+			clientId,
```

Comment:
Any need for `clientId` to be here?

Reply:
Yep. As a rule, you need to put all the variables you're using inside the `useEffect` hook into `useEffect`'s dependency array.